homeassistant/components/dio_chacon/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for dio_chacon integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from dio_chacon_wifi_api import DIOChaconAPIClient
+from dio_chacon_wifi_api.exceptions import DIOChaconAPIError, DIOChaconInvalidAuthError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_PASSWORD, CONF_UNIQUE_ID, CONF_USERNAME
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DioChaconConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for dio_chacon."""
+
+    _username: str
+    _password: str
+    _user_id: str
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        if user_input is not None:
+            self._username = user_input[CONF_USERNAME]
+            self._password = user_input[CONF_PASSWORD]
+
+            errors: dict[str, str] = {}
+            try:
+                # Validate the user name and password and retrieve the technical user id.
+                client = DIOChaconAPIClient(self._username, self._password)
+                self._user_id = await client.get_user_id()
+                await client.disconnect()
+            except DIOChaconAPIError:
+                errors["base"] = "cannot_connect"
+            except DIOChaconInvalidAuthError:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+
+            if errors:
+                return self._show_setup_form(errors)
+
+            entry_id: str = "dio_chacon_" + self._user_id
```

Comment:
```suggestion
            except Exception:  # pylint: disable=broad-except
                _LOGGER.exception("Unexpected exception")
                errors["base"] = "unknown"

            else:
                entry_id: str = "dio_chacon_" + self._user_id
```
This way you don't need the extra function to show the form because you're only calling it once

Reply:
ok corrected it and removed the function.